src/processes/follow-topic-process.js

```diff
@@ -191,16 +191,22 @@ const followTopicProcess = async (job, done) => {
         return;
       }
 
+      // use dummy sender user id if provided (for testing purpose)
+      const senderUserId = data?.dummy_sender_user_id || communityMessageFormat.user_id;
       // check if user is anonymous
-      const [senderUser] = await Promise.all([User.findByPk(communityMessageFormat.user_id)]);
+      const [senderUser] = await Promise.all([User.findByPk(senderUserId)]);
+      const communityMessageFormatObject = {
+        user_id: senderUserId,
+        message: communityMessageFormat.message
+      };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There is a potential issue with the use of optional chaining (`?.`) on `data` when assigning `senderUserId`. If `data` is `null` or `undefined`, `senderUserId` will be `undefined`, which could lead to unexpected behavior in subsequent code. Consider adding a null check for `data` before this line.
```diff
- const senderUserId = data?.dummy_sender_user_id || communityMessageFormat.user_id;
+ if (!data) {
+   throw new Error('Data is not provided');
+ }
+ const senderUserId = data.dummy_sender_user_id || communityMessageFormat.user_id;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->